manuscript/chapter3.txt

```diff
@@ -7,29 +7,29 @@ C> By [John Haynes](https://www.youtube.com/channel/UCJVXolxwB4x3EsBAzSACCTg) |
+Metaphorically, a password is a key that fits into and unlocks a particular lock. Only you have the key, but anyone can come up and inspect the lock. With a mechanical lock, it is difficult to determine what the internal dimensions of the lock without specialized tools like lock picks. Someone proficient in the use of lockpicks could potentially determine the depth of each pin and use that information to cut a key that would unlock the lock.
 
-The same sort of concept is true for passwords. Each password should have a unique algorithmic hash. To obtain a hash, a complex mathematical algorithm is run against a string of data and the output is an extremely unique character string. For some weaker hash algorithms, there have been hash collisions where two different sets of data have resulted in the same outputted hash. However, when considering human-generated passwords, it is normally not necessary to worry about hash collisions. It is sufficient to say that if you have the hash of a password you have the password in an encrypted state. The password hash is how the password is stored on any modern operating system like Windows, macOS, or Linux or for encrypted containers like BitLocker or encrypted 7-Zip files. With the right tools, that is the only part of the password that will be available for an examiner to inspect, just like the mechanical part of a lock is the only thing to inspect on a locked door if someone were to try and pick the lock. There are methods to prevent the extraction of a password hash, but it is reasonable to attempt to find a method to extract a hash from a system if the individual has physical access to the electronic device, encrypted file, or a forensic image (.E01, dd, or similar) of an encrypted volume or file.
+In this metaphor, the password is the key and its "hash value" is the internal dimensions of the lock. To obtain a hash, a complex mathematical algorithm is run against a string of data and the output is an extremely unique character string. Some weaker hash algorithms can produce the same hash for two different sets of data. This is not typically an issue with hashed passwords, and it is safe to say that if you have a hash of a password you have the password in an encrypted state. The password hash is how the password is stored on any modern operating system like Windows, macOS, or Linux or for encrypted containers like BitLocker or encrypted 7-Zip files. With proper security practices, the hash is the only part of the password that will be available for an examiner to inspect, just as a lockpicker might inspect a lock. If you have access to the encrypted item itself, whether it is a physical device, a forensic image (e.g. a .E01 or .dd), or an individual encrypted file file, it is reasonable to attempt to extract the hash value.
```

Comment:
Minor typo correction, remove the duplicate word (file): 

"...a forensic image (e.g. a .E01 or .dd), or an individual encrypted file, it is reasonable to attempt to extract the hash value."